langchain_dev.py

```diff
@@ -0,0 +1,171 @@
+import os
+import json
+from diarize import get_token
+from pathlib import Path
+from typing import Any
+from langchain_community.llms import HuggingFaceEndpoint
+from langchain.chains import LLMChain
+from langchain.prompts import PromptTemplate
+from langchain import FewShotPromptTemplate
+from langchain_community.document_loaders import WebBaseLoader
+from langchain_community.document_loaders import TextLoader
+from langchain_community.embeddings.sentence_transformer import (
+    SentenceTransformerEmbeddings,
+)
+from langchain_community.vectorstores import Chroma
+from langchain_text_splitters import TokenTextSplitter, CharacterTextSplitter, RecursiveCharacterTextSplitter
+from langchain import hub
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.runnables import RunnablePassthrough
+from langchain.chains import RetrievalQA
+from langchain.memory import ConversationBufferMemory
+from langchain.chains import ConversationalRetrievalChain
+
+
+os.environ["HUGGINGFACEHUB_API_TOKEN"] = get_token()
+
+def llm_init(repo_id: str = "google/gemma-7b") -> Any:
+    # Set up a Hugging Face Endpoint for Gemma 2b model
+    llm = HuggingFaceEndpoint(
+    repo_id=repo_id, max_length=1024, temperature=0.1
+    )
+
+    return llm 
+
+def json_to_text(json_path: Path, output_path: Path) -> None: 
+    with open(json_path, 'r') as f: 
+        json_obj = json.load(f)
+
+    with open(output_path, 'w', encoding = 'utf-8') as f: 
+        for speaker_info in json_obj: 
+            text = ""
+            text = f"[{speaker_info['start']} : {speaker_info['end']}] {speaker_info['speaker']}: {speaker_info['text']}\n"
+            f.write(text)
+
+
+def few_shot_prompts():
+    # TODO: Work this to figureout a way to add this to the conversational chain.
+    # Define examples that include user queries and AI's answers specific to Kaggle competitions
+    examples = [
```

Comment:
What are these examples? Few shot prompts? How is this related to our tasks? I thought we had a summarization task?

Reply:
I will update the new code now. But essentially it's for the RAG pipeline. I went to sleep before finishing that.